lib/nib/history/compose.rb

```diff
@@ -0,0 +1,45 @@
+require 'tempfile'
+require 'tmpdir'
+
+class Nib::History::Compose
+  attr_reader :dir, :volume_name
+
+  def initialize
+    @volume_name = 'nib_history'
+    @dir = "#{Dir.tmpdir}/#{Dir.pwd.split('/').last}"
+
+    FileUtils.mkdir_p(dir)
+  end
+
+  def path
+    file.path
+  end
+
+  private
+
+  def config
+    @config ||= YAML.safe_load(`docker-compose config`)
+  end
+
+  def file
+    @file ||= Tempfile.open('compose', dir) do |compose|
+      compose.write(extended.to_yaml)
+      compose
+    end
+  end
+
+  def extended
+    config.dup.tap do |h|
```

Comment:
Can this method be refactored some?  I have to scratch my head to follow what's happening here.